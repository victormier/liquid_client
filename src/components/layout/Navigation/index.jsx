import React from 'react';
import { Link } from 'react-router';
import { LinkContainer } from 'react-router-bootstrap';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import PropTypes from 'prop-types';

const Navigation = props => (
  <Navbar collapseOnSelect>
    <Navbar.Header>
      <Navbar.Brand>
        <Link to="/">
          {props.title}
        </Link>
      </Navbar.Brand>
      <Navbar.Toggle />
    </Navbar.Header>
    <Navbar.Collapse>
      <Nav>
        {props.links.map(item => (<LinkContainer key={item.id} to={item.href}>
          <NavItem>{item.label}</NavItem>
        </LinkContainer>)
        )}
      </Nav>
    </Navbar.Collapse>
  </Navbar>);

Navigation.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      href: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default Navigation;
