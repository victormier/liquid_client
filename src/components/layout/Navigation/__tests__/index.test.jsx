import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Navigation from '../index';


describe('component', () => {
  const containerProps = {
    title: 'Navigation title',
    links: [
      {
        id: 1,
        href: 'Link 1 href',
        label: 'Link 1 label',
      },
      {
        id: 2,
        href: 'Link 2 href',
        label: 'Link 2 label',
      },
    ],
  };

  it('renders with all props', () => {
    const container = shallow(
      <Navigation {...containerProps} />
    );

    expect(shallowToJson(container)).toMatchSnapshot();
  });
});
