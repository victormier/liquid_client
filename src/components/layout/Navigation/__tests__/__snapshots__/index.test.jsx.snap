// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders with all props 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  collapseOnSelect={true}
>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/"
      >
        Navigation title
      </Link>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <LinkContainer
        action="push"
        onlyActiveOnIndex={false}
        to="Link 1 href"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Link 1 label
        </NavItem>
      </LinkContainer>
      <LinkContainer
        action="push"
        onlyActiveOnIndex={false}
        to="Link 2 href"
      >
        <NavItem
          active={false}
          disabled={false}
        >
          Link 2 label
        </NavItem>
      </LinkContainer>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
